app/src/main/java/com/iic/shopingo/api/Request.java

```diff
@@ -10,45 +19,56 @@
  */
 public class Request {
   public static OutgoingRequestAPIResult makeRequest(List<String> shoppingItems, int offer, String shopperId) {
-    ServerSim.simulateRequest();
-    return new OutgoingRequestAPIResult(new OutgoingRequest("12345"));
+    try {
+      Server server = new Server(CurrentUser.getInstance().userInfo.getUid());
+      HashMap<String, Object> params = new HashMap<>();
+      params.put(Constants.Parameters.REQUESTS_SHOPPER_ID, shopperId);
+      params.put(Constants.Parameters.REQUESTS_OFFER, offer);
+      params.put(Constants.Parameters.REQUESTS_ITEMS, shoppingItems);
+      ApiOutgoingRequest response = server.post(Constants.Routes.REQUESTS_CREATE_PATH, ApiOutgoingRequest.class, params);
+      return new OutgoingRequestAPIResult(response);
+    } catch (IOException e) {
+      return new OutgoingRequestAPIResult(e.getMessage());
+    }
   }
 
   public static APIResult cancelRequest() {
-    ServerSim.simulateRequest();
-    return new APIResult();
+    try {
+      Server server = new Server(CurrentUser.getInstance().userInfo.getUid());
```

Comment:
Each API call creates a new Server object which in turns create a new HttpClient.
It's better to have a single instance of a Server that is initialized after the user logs in, and make all the API calls through it.